circuit_maintenance_parser/parsers/turkcell.py

```diff
@@ -0,0 +1,69 @@
+"""Turkcell parser."""
+import logging
+import re
+from typing import Dict
+
+from bs4.element import ResultSet  # type: ignore
+from dateutil import parser
+
+from circuit_maintenance_parser.errors import ParsingError
+from circuit_maintenance_parser.parser import Html, Impact, CircuitImpact, Status
+
+# pylint: disable=too-many-nested-blocks, too-many-branches
+
+logger = logging.getLogger(__name__)
+
+
+class HtmlParserTurkcell1(Html):
+    """Notifications Parser for Turkcell notifications."""
+
+    def parse_html(self, soup, data_base):
+        """Execute parsing."""
+        data = data_base.copy()
+        try:
+            self.parse_tables(soup.find_all("table"), data)
+            return [data]
+
+        except Exception as exc:
+            raise ParsingError from exc
+
+    def parse_tables(self, tables: ResultSet, data: Dict):
+        """Parse tables."""
+        # Main table
+        td_elements = tables[0].find_all("td")
+        for idx, td_element in enumerate(td_elements):
+            if "Dear Customer" in td_element.text.strip():
+                if "planned" in td_element.text.strip():
+                    data["status"] = Status["CONFIRMED"]
+                else:
+                    data["status"] = Status["CONFIRMED"]
+            if "Maintenance Number" in td_element.text.strip():
+                data["maintenance_id"] = td_elements[idx + 1].text.strip()
+            elif "Start" in td_element.text.strip():
+                data["start"] = self.dt2ts(parser.parse(td_elements[idx + 1].text.strip()))
+            elif "End" in td_element.text.strip():
+                data["end"] = self.dt2ts(parser.parse(td_elements[idx + 1].text.strip()))
+            elif "Impact of the maintenance" in td_element.text.strip():
+                data["summary"] = td_elements[idx + 1].span.text.strip()
+                if len(tables) == 1:
+                    data["circuits"] = []
+                    p_elements = td_elements[idx + 1].find_all("p")
+                    for element in p_elements:
+                        # Example match:
+                        #   Eth-Trunk1.1               up      up       111111111111111|01-CUSTOMER|LOCATION|LINK
+                        if re.match(r".+[ \t]([0-1]+\|.+\|.+\|.+)", element.text.strip()):
+                            groups = re.search(r".+[ \t]([0-1]+\|.+\|.+\|.+)", element.text.strip())
+                            if groups:
```

Comment:
Doesn't seem like the initial `re.match` call is needed - redundant?

```suggestion
                        groups = re.search(r".+[ \t]([0-1]+\|.+\|.+\|.+)", element.text.strip())
                        if groups:
```